pglookout/pglookout.py

```diff
@@ -813,10 +814,22 @@ def delete_alert_file(self, filename):
     def within_dbpoll_time(self, time1, time2):
         return abs((time1 - time2).total_seconds()) < self.config.get("db_poll_interval", 5.0)
 
+    def _check_cluster_monitor_thread_health(self, now: float) -> None:
+        health_timeout_seconds = self.config.get("cluster_monitor_health_timeout_seconds")
+        if health_timeout_seconds:
+            last_run_time = self.cluster_monitor.last_monitoring_loop_start_time or self._start_time
+            # last_run_time can only be None if main_loop or this function is called directly
+            if last_run_time is not None:
+                seconds_since_last_run = now - last_run_time
+                if seconds_since_last_run >= health_timeout_seconds:
+                    self.stats.increase("cluster_monitor_health_timeout")
+                    self.log.warning("cluster_monitor has not been running for %.1f seconds", seconds_since_last_run)
+
     def main_loop(self):
         while self.running:
             try:
                 self.check_cluster_state()
+                self._check_cluster_monitor_thread_health(now=time.monotonic())
```

Comment:
The variable is called `last_monitoring_loop_start_time`, but the measurement is not done from the start of the loop, it's done only after the check has succeeded. This also does not match the documentation.

Updating the time only after a successful check is a good thing:
Strictly speaking "does the loop still run" and "can we successfully check the cluster state" are two different problems, but in the end any alerting would check both and alert the same person to take a look at what's wrong, so there wouldn't be much point in emitting two separate metrics (and we already have the `unexpected_exception` metric to drill down into details once the `cluster_monitor_health_timeout` raises).

I think `last_monitoring_loop_start_time` should be `last_monitoring_success_time` and the documentation updated accordingly.